### DISCUSSION TITLE
unable to hit stt v2 recognizer ( NOT_FOUND: Unable to find Recognizer )

### DISCUSSION THREAD

**Discussion by swaman** [googleapis/google-cloud-cpp#14688] [Q&A]
Below is my sample code, whenever i running this code getting below errors--->
Recognize stream finished with an error: NOT_FOUND: Unable to find Recognizer 

sample code --->

#include "google/cloud/speech/v2/speech_client.h"
#include "google/cloud/project.h"
#include <iostream>
#include <fstream>
#include <string>
#include <thread>
#include "google/cloud/common_options.h"
namespace speech = google::cloud::speech_v2;
using namespace ::google::cloud::speech::v2;
using RecognizeStream = ::google::cloud::AsyncStreamingReadWriteRpc<StreamingRecognizeRequest,StreamingRecognizeResponse>;
using namespace std;
auto constexpr kUsage = R"""(Usage:
  streaming_transcribe [--bitrate N] audio.(raw|ulaw|flac|amr|awb)
)""";

// Write the audio in 64k chunks at a time, simulating audio content arriving
// from a microphone.
void MicrophoneThreadMain(RecognizeStream& stream,std::string const& file_path) {
	StreamingRecognizeRequest request;
	std::ifstream file_stream(file_path, std::ios::binary);
	auto constexpr kChunkSize = 32 * 1024;
	std::vector<char> chunk(kChunkSize);
	while (true) {
		// Read another chunk from the file.
		file_stream.read(chunk.data(), chunk.size());
		auto const bytes_read = file_stream.gcount();
		// And write the chunk to the stream.
		if (bytes_read > 0) {
			request.set_audio(chunk.data(), bytes_read);
			//std::cout << "Sending " << bytes_read / 1024 << "k bytes." << std::endl;
			if (!stream.Write(request, grpc::WriteOptions()).get())
			{
				break;
			}else
			{
				std::cout << "Sending " << bytes_read / 1024 << "k bytes." << std::endl;
			}
		}
		if (!file_stream) {
			// Done reading everything from the file, so done writing to the stream.
			stream.WritesDone().get();
			cout<<"done... wrting data\n";
			break;
		}
		// Wait a second before writing the next chunk.
		std::this_thread::sleep_for(std::chrono::seconds(1));
	}
}
void ConfigureRecognizer(google::cloud::speech::v2::RecognizeRequest& request) {
	*request.mutable_config()->add_language_codes() = "hi-IN";
	request.mutable_config()->set_model("long");
	*request.mutable_config()->mutable_auto_decoding_config() = {};
}
struct location {
   using Type = std::string;
 };

int main(int argc, char** argv) try {
	// Create a Speech client with the default configuration
	//auto options = ::google::cloud::Options{}.set<::google::cloud::EndpointOption>("asia-south1-speech.googleapis.com");
 // auto client = google::cloud::speech_v2::SpeechClient(
      //google::cloud::speech_v2::MakeSpeechConnection(options));
	//auto client = speech::SpeechClient(speech::MakeSpeechConnection(options));

	// Parse command line arguments.
	//auto args = ParseArguments(argc, argv);
	auto const file_path = argv[1];
	cout<<"file path= "<< file_path<<"\n";
	StreamingRecognizeRequest request;
	//request.set_recognizer("projects/sr-project-eva-prod/locations/global/recognizers/gcprecogtest");
	auto& streaming_config = *request.mutable_streaming_config();
	//*streaming_config.mutable_config() = args.config;
	std::shared_ptr<speech::SpeechConnection> connection;
	google::cloud::speech::v2::RecognizeRequest rec_req;
	ConfigureRecognizer(rec_req);
	rec_req.set_recognizer("projects/myproject/locations/asia-south1/recognizers/myrecognizer");

	*streaming_config.mutable_config() = *rec_req.mutable_config();
	
	::google::cloud::Options opt;
	opt.set<location>("asia-south1");
	auto client = speech::SpeechClient(speech::MakeSpeechConnection(opt));

	// Begin a stream
	//auto client = speech::SpeechClient(speech::MakeSpeechConnection(opt));
	auto stream = client.AsyncStreamingRecognize();
	// The stream can fail to start, and `.get()` returns an error in this case.
	if (!stream->Start().get()) throw stream->Finish().get();
	// Write the first request, containing the config only.
	if (!stream->Write(request, grpc::WriteOptions{}).get()) {
		// Write().get() returns false if the stream is closed.
		throw stream->Finish().get();
	}
	// Simulate a microphone thread using the file as input.
	auto microphone =
		std::thread(MicrophoneThreadMain, std::ref(*stream), file_path);
	// Read responses.
	auto read = [&stream] {
		std::cout<<"reading response\n";
		return stream->Read().get();
	};
	for (auto response = read(); response.has_value(); response = read()) {
		// Dump the transcript of all the results.
		for (auto const& result : response->results()) {
			std::cout << "Result stability: " << result.stability() << "\n";
			for (auto const& alternative : result.alternatives()) {
				std::cout << alternative.confidence() << "\t"
					<< alternative.transcript() << "\n";
			}
		}
	}
	auto status = stream->Finish().get();
	microphone.join();
	if (!status.ok()) throw status;
	return 0;
} catch (google::cloud::Status const& s) {
	std::cerr << "Recognize stream finished with an error: " << s << "\n";
	return 1;
} catch (std::exception const& ex) {
	std::cerr << "Standard C++ exception thrown: " << ex.what() << "\n"
		<< kUsage << "\n";
	return 1;
}

**Comment by scotthart** [comment]
It appears you set the recognizer:
```
rec_req.set_recognizer("projects/myproject/locations/asia-south1/recognizers/myrecognizer");
```
but, I don't see any code to create a recognizer with that name. Try using the implicit recognizer:
```
rec_req.set_recognizer("projects/myproject/locations/asia-south1/recognizers/_");
```
You may also want to use `GetRecognizer` or `ListRecognizers` to verify that any non-implicit recognizers you want to use exist.
